core/kernel/persistence/smoothsql/class.SmoothRdf.php

```diff
@@ -58,27 +58,79 @@ public function get($subject, $predicate)
     {
         throw new \common_Exception('Not implemented');
     }
-    
+
     /**
      * (non-PHPdoc)
      * @see \oat\generis\model\data\RdfInterface::add()
      */
-    public function add(\core_kernel_classes_Triple $triple)
+    public function add(core_kernel_classes_Triple $triple)
     {
-        $query = "INSERT INTO statements ( modelId, subject, predicate, object, l_language, epoch, author) VALUES ( ? , ? , ? , ? , ? , ?, ?);";
-        $success = $this->getPersistence()->exec($query, [$triple->modelid, $triple->subject, $triple->predicate, $triple->object, is_null($triple->lg) ? '' : $triple->lg, $this->getPersistence()->getPlatForm()->getNowExpression(), is_null($triple->author) ? '' : $triple->author]);
-        if ($triple->predicate == OntologyRdfs::RDFS_SUBCLASSOF || $triple->predicate == OntologyRdf::RDF_TYPE) {
-            $eventManager = $this->model->getServiceLocator()->get(EventManager::SERVICE_ID);
-            $eventManager->trigger(new ResourceCreated($this->model->getResource($triple->subject)));
+        $query = "INSERT INTO statements ( modelId, subject, predicate, object, l_language, epoch, author) "
+            . "VALUES ( ? , ? , ? , ? , ? , ?, ?);";
+
+        $success = $this->getPersistence()->exec(
+            $query,
+            [
+                $triple->modelid,
+                $triple->subject,
+                $triple->predicate,
+                $triple->object,
+                is_null($triple->lg) ? '' : $triple->lg,
+                $this->getPersistence()->getPlatForm()->getNowExpression(),
+                is_null($triple->author) ? '' : $triple->author
+            ]
+        );
+
+        if ($success > 0) {
+            $this->checkResourceTriggers($triple);
         }
+
         return $success;
     }
+
+    /**
+     * @inheritDoc
+     */
+    public function addTripleCollection(iterable $triples)
+    {
+        $valuesToInsert = [];
+
+        foreach ($triples as $triple) {
+            $valuesToInsert [] = $triple;
+
+            if (count($valuesToInsert) >= self::BATCH_SIZE) {
+                $this->insertTriples($valuesToInsert);
+                $valuesToInsert = [];
+            }
+        }
+
+        if (!empty($valuesToInsert)) {
+            $this->insertTriples($valuesToInsert);
+        }
+    }
+
+    protected function insertTriples(array $triples)
+    {
+        $values = array_map([$this,"tripleToValue"], $triples);
+        $isInsertionSuccessful = $this->insertValues($values);
```

Comment:
In case this `insertValues` will fail will it throw any errors? Should we used try block in order to ensure that we successfully performed query execution?